### PR TITLE
[Android] Call callback onFinish() immediately when trying to move camera to current position

### PR DESCRIPTION
**Pull request by philemonmerlet**
Camera movement methods in Transform class allow the caller to pass a callback, to know when action is performed. But if the target movement is the same as the current position, the methods end and the callback is not called at all.
Of course the caller could check himself if the movement has an effect, but it seems strange to perform this check at 2 different levels. Also, the exact check method isValidCameraPosition is private to the Transform class, and we have no guarantee that the conditions will stay the same in the future.

The simple solution suggested here is to call onFinish immediately in the Transform class when the requested movement is not valid. Then the caller is sure that he will receive a callback, whatever the call result is.

### CODE REVIEW

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java [hunk 1]
```diff
@@ -120,6 +120,8 @@ public void run() {
           }
         }
       });
+    } else if(callback != null) {
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java]: missing space after `if`

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java [hunk 2]
```diff
@@ -137,6 +139,8 @@ final void easeCamera(@NonNull MapboxMap mapboxMap, CameraUpdate update, int dur
       mapView.addOnCameraDidChangeListener(this);
       nativeMap.easeTo(cameraPosition.target, cameraPosition.zoom, cameraPosition.bearing, cameraPosition.tilt,
         cameraPosition.padding, durationMs, easingInterpolator);
+    } else if(callback != null) {
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java]: missing space after if

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java [hunk 3]
```diff
@@ -157,6 +161,8 @@ public final void animateCamera(@NonNull MapboxMap mapboxMap, CameraUpdate updat
       mapView.addOnCameraDidChangeListener(this);
       nativeMap.flyTo(cameraPosition.target, cameraPosition.zoom, cameraPosition.bearing,
         cameraPosition.tilt, cameraPosition.padding, durationMs);
+    } else if(callback != null) {
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java]: missing space after if

**Comment by philemonmerlet** [comment]
Thanks for your review. I just added a test in TransformTest class, is it ok for you ?

**Comment by tobrun** [comment]
@philemonmerlet looks good, could you add 2 more test to validate that we are exercising the full api? A test was added for jumpTo but not easeTo/animateTo. Thanks again for contributing!

**Comment by philemonmerlet** [comment]
No problem, I'll look into this. 
I just realized that other callback calls are made through handler.post, should I change the code to match this behavior or is it ok to call it directly ?

**Comment by julianrex** [comment]
@tobrun FYI I have https://github.com/mapbox/mapbox-gl-native/pull/15473 in flight too.

**Comment by tobrun** [comment]
@philemonmerlet looks good, note that we recently revisited to not run CI builds for external contributors due to security reasons. I will have to cherry pick your commits in another PR. Thanks again for your contribution!

**Comment by tobrun** [comment]
following up on this in https://github.com/mapbox/mapbox-gl-native/pull/15558

**Pull request closed without merging**